apps/stable_diffusion/src/models/opt_params.py

```diff
@@ -13,6 +16,14 @@
     "CompVis/stable-diffusion-v1-4": ["stablediffusion", "v1_4"],
 }
 
+# TODO: Implement a better way to handle the KeyNotFound error by delaying the invocation.
+variant = None
+version = None
+try:
+    variant, version = hf_model_variant_map[args.hf_model_id]
+except:
+    pass
+
```

Comment:
This has to be local to each function, i.e., means inside get_unet(), get_vae() and get_clip(), as these functions can be called multiple times with different model ids.